docs: Update FAQs with IAM role answer (#3093)
hashicorp/boundary: website/content/docs/troubleshoot/faq.mdx

[0.4 announcement](https://www.hashicorp.com/blog/announcing-hashicorp-boundary-0-4) and there are ample
[tutorials](/boundary/tutorials/access-management/oss-vault-cred-brokering-quickstart) of this scenario.
2. Boundary can use Vault as an OIDC provider to enable sign-in with Vault's supported auth methods (even non-OIDC auth methods like Active Directory kerberos/LDAP).
This scenario is walked through in this [tutorial](/vault/tutorials/auth-methods/oidc-identity-provider).
3. OSS Boundary can use Vault as the external KMS that serves as Boundary's root of trust. More information on this use case can be found
[here](/boundary/docs/configuration/kms/transit).

#### Q: What Identity Providers does Boundary Support?
Boundary supports [OIDC authentication](/boundary/tutorials/access-management/oidc-auth), which allows support for many popular IdPs
like Okta, Azure AD, Auth0, etc. For non-OIDC authentication protocols (such as LDAP), users can also log in with an OIDC bridge identity provider
such as [Vault's OIDC bridge](/vault/docs/concepts/oidc-provider) capabilities released in 1.9, [Dex](https://dexidp.io), or various others.

> #### Q: Does Boundary support Multi-factor Authentication (MFA)

Reviewer: ```suggestion
#### Q: Does Boundary support Multi-factor Authentication (MFA)?
```

- [Dynamic Host Catalogs on Azure](/boundary/tutorials/access-management/azure-host-catalogs)

#### Q: Session management capabilities - Does Boundary support live session monitoring and termination?
**Session Logging/Monitoring:** Supported. Boundary creates a session log of all sessions created between identities and targets that have been onboarded to
Boundary. You can learn how to monitor these sessions in this [tutorial](/boundary/tutorials/getting-started/getting-started-connect#manage-sessions).
Boundary supports audit logs for [Boundary OSS](/boundary/tutorials/oss-configuration/event-logging) and [audit log streaming](/hcp/docs/boundary/audit-logging)
for HCP Boundary. Audit logs for both distributions can be exported to SIEM or BI tools.

**Session Termination:** Supported. Session termination for Boundary administrators is a supported capability, as demonstrated in this
[tutorial](/boundary/tutorials/getting-started/getting-started-connect#manage-sessions).

> #### Q: As an AWS user, can I use IAM role to configure Boundary's dynamic host catalog?

Reviewer: ```suggestion
#### Q: As an AWS user, can IAM roles be used to configure Boundary's dynamic host catalog?
```

> At this time, it is not possible to configure Boundary's dynamic host catalog using IAM role.
> You can configure the dynamic host catalog using IAM user, however.
> For more information, refer to the [Dynamic host catalogs on AWS](/boundary/tutorials/access-management/aws-host-catalogs) tutorial.

Reviewer: hmm, wondering if IAM roles/users needs to be plural or singular? I hear it most often as "IAM roles" or "an IAM role"

```suggestion
At this time, it is not possible to configure Boundary's dynamic host catalog using IAM roles.
You can configure the dynamic host catalog using an IAM user, however.
For more information, refer to the [Dynamic host catalogs on AWS](/boundary/tutorials/access-management/aws-host-catalogs) tutorial.
```
